[Woo POS][App Listings] ASC promo screenshots

iamgabrielma opened the pull request:
Closes WOOMOB-1644

## Description

This PR handles creating the new POS promotional screenshots and submitting them to the App Store through our existing fastlane automation. 

## Changes
- Added backgrounds and frames for landscape images
- Updated `fastfile` to include the new mapping `app_store_screenshot` -> `promo_screenshot`
- Updated `screenshots.json` to include a new landscape device, mostly mapping the existing portrait one. These are mounted programmatically by pixel/pixel-offset, but there is no references anywhere, so was a bit of iteration to make them fit adequately:

<img width="934" height="439" alt="Screenshot 2025-12-16 at 11 56 42" src="https://github.com/user-attachments/assets/e968d881-2155-4b44-a542-2e7685b11c85" />

- iPad
  - Runs UI POS screenshot tests on iPads only
  - Intersects the new screenshots in the existing ones (mostly a dependency from UI tests having to be run in a given order) 
  - Mounts background/frame/screenshot/text for the promotional screenshot

- iPhones
  - Moves existing 1-5 iPhone screenshots to default folder
  - Updates which text to image mapping to use for 4th and 5th screenshots, due the new inclusion of iPad screenshots since we don't have the same number of them in each device now and would map the wrong text.


## Test Steps
- Run `WooCommerceScreenshots.testScreenshots()` both in iPad and iPhone as your run destination, they should pass (some UI tests are a bit flaky, but on a second run they should normally pass correctly)
- The screenshot output cannot be tested locally, but we can find the artifacts from the CI job in the latest job `https://github.com/woocommerce/woocommerce-ios/actions/runs/20289167160/job/58286405158?pr=16405` under `Process screenshots` > Archive promo screenshots > Download the artifacts to see all screenshots for all locales (the untranslated screenshots will show a placeholder rather than their text)

<img width="1292" height="623" alt="Screenshot 2025-12-18 at 10 56 17" src="https://github.com/user-attachments/assets/99611eb0-f396-42bf-944a-87f5ff3c0802" />

Example for Spanish, where the POS appears in Spanish locale, and the metadata text is just untranslated placeholder:

<img width="361" height="279" alt="Screenshot 2025-12-18 at 10 57 50" src="https://github.com/user-attachments/assets/2e560f87-856a-4fe6-8111-b795ffc370bb" />


---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

iamgabrielma committed:
add all app_store_screenshots to default

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/262">23.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
runs UI POS screenhot tests on iPads only

iamgabrielma committed:
use-mocked-card-present-payment for UI tests

Despite tests passing, the contents of the pos-payment screenshot was incorrect, as lands on the failure case for `checkOrderIsStillEligibleForPayment` when attempting to retrieve the order details, hence showing a “Error: unable to process payment. we could not fetch the latest order details” rather than the payment view.

Despite passing the CPP mock in the POSTabCoordinator, we were never setting it in the test as a launch argument.

iamgabrielma committed:
reorder screenshots to pass create_promo lane

Since we run the notification screenshot the last one due to the UI test flow, when we run the create_promo lane then screenshot descriptions are miss-aligned with what they show which fails the job.

iamgabrielma committed:
update screenshot variant

iamgabrielma committed:
reorder screenshots following UI test order

At the moment we intersect POS screenshots, but “product-add” screenshot happens as the 7th, while in screenshots.js is set to be the 4rth. This fails the create_promo job as cannot find the correct index along the screenshot name

iamgabrielma committed:
Merge branch 'trunk' into task/WOOMOB-1644-generate-pos-promo-screenshots

iamgabrielma committed:
add new device in landscape to screenshots

initial attempt to set the screenshot landscape composition programmatically. might need iteration after checking the result artifacts

iamgabrielma committed:
update screenshots lane with new translatable text

iamgabrielma committed:
Adjust landscape text and frame offsets

iamgabrielma committed:
Resolve text to image wrong mapping for iPhone devices

iPhone screenshots 4 and 5 are using the wrong text files because new POS screenshots are inserted into the sequence

iamgabrielma commented:
For some reason, marking the PR as ready seems to have triggered the screenshots job again, ~~so we may need to wait 3-4h until it completes to check the artifacts 🤦 (if we're lucky, the previous job artifacts won't be expired/deleted but stay there for a while)~~

Edit: Latest screenshots can be found here: https://github.com/woocommerce/woocommerce-ios/actions/runs/20289167160/job/58286405158?pr=16405

joshheald reviewed (approved):
Well done for wrangling this, screenshots look good
